src/hardforks/hardforks.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2014-2019, The Monero Project
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification, are
+// permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#pragma once
+
+#include <stdint.h>
+#include <time.h>
+#include "cryptonote_basic/hardfork.h"
```

Comment:
I think it's better to have the hard fork data class defined under the `hardforks` library which is totally independent and easily usable for 3rd party developers. Including `cryptonote_basic/hardfork.h` makes it dependent on `cryptonote_basic.h` for no good reason.

Reply:
Well, cryptonote_basic.h would then have to depend on hardforks.h, no ?
It'd be like having std::vector depending on a file that defines a particular vector, it seems backwards.

Reply:
> Well, cryptonote_basic.h would then have to depend on hardforks.h, no ?

Yes, the right way IMO is to have `libhardforks` as an absolutely independent library (similar to `libversion`) and let other apps needing to fetch the data link to it. I don't see any problem in `cryptonote_basic/hardfork.h` having to include `hardforks/hardforks.h` (FWIW `cryptonote_basic.h` doesn't need `hardforks.h`).

Your current design will force 3rd party developers wanting to incorporate the hardfork data to their project to copy all dependent header files stemming from `cryptonote_basic/hardfork.h` such as `syncobj.h`, `cryptonote_basic.h`, `serialization/variant.h`, etc, which are completely unrelated.

>  It'd be like having std::vector depending on a file that defines a particular vector, it seems backwards.

I don't understand this analogy.`std::vector` is a class template, seems unrelated.

Reply:
The actual precise type is not relevant, but I've done that, but I also switched the type so it gets defined in the included file for it not to feel wrong.